odoo/addons/base/models/res_users.py

```diff
@@ -1728,10 +1741,31 @@ def change_password_button(self):
         for line in self:
             if not line.new_passwd:
                 raise UserError(_("Before clicking on 'Change Password', you have to write a new password."))
-            line.user_id.write({'password': line.new_passwd})
+            line.user_id._change_password(line.new_passwd)
         # don't keep temporary passwords in the database longer than necessary
         self.write({'new_passwd': False})
 
+class ChangePasswordOwn(models.TransientModel):
+    _name = "change.password.own"
+    _description = "User, change own password wizard"
+
+    new_password = fields.Char(string="New Password")
+    confirm_password = fields.Char(string="New Password (Confirmation)")
+
+    @api.constrains('new_password', 'confirm_password')
+    def _check_password_confirmation(self):
+        if self.confirm_password != self.new_password:
+            raise ValidationError(_("The new password and its confirmation must be identical."))
+
+    @check_identity
+    def change_password(self):
+        self.env.user._change_password(self.new_password)
+        self.write({'new_password': False, 'confirm_password': False})
+        self.unlink()
```

Comment:
Why do you erase the new password before removing the the row completely ? Is it to remove the data from the transaction cache ?

Reply:
I think it's just an accumulation: initially I blanked the fields figuring "once the password is changed we should remove the new passwords for safety", similar to the bulk wizard.

Then I figured "I could just remove the entire record", no cause to create a new row in the database.

Then I figured "This is very dumb just lock down access so only the creator can see it".

But removed none of the previous safeties. I should probably still keep either (probably the removal, unless that has worse database-level implications @odony ?) and set `_transient_max_hours` to 0.1: since the action opens a wizard in creation mode, it should only be created the first time the user actually triggers the "change password" action, worse case scenario there's a validation error and the user needs to update values, that should fit in 6 minutes or so.

Reply:
one of the two should be enough, indeed